cpp/include/cudf/legacy/io_types.hpp

```diff
@@ -112,58 +127,93 @@ struct avro_read_arg {
+  cudf::size_type header = 0;       ///< Header row index, zero-based counting;
+                                    ///< default is no header reading
+
+  std::vector<std::string> names;  ///< Names of the columns
+  std::vector<std::string>
+      dtype;  ///< Data types of the column; empty to infer dtypes
+
+  std::vector<int>
+      use_cols_indexes;  ///< Indexes of columns to read; empty is all columns
+  std::vector<std::string>
+      use_cols_names;  ///< Names of column to read; empty is all columns
+
+  std::vector<std::string>
+      true_values;  ///< Values to recognize as boolean True; default empty
+  std::vector<std::string>
+      false_values;  ///< Values to recognize as boolean False; default empty
+  std::vector<std::string>
+      na_values; /**< Values to recognize as invalid; default values:
+                 '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN',
+                 '-nan', '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL',
+                 'NaN', 'n/a', 'nan', 'null'. */
+  bool keep_default_na = true;  ///< Keep the default NA values
+  bool na_filter =
+      true;  ///< Detect missing values (empty strings and the values in
+             ///< na_values); disabling can improve performance
+
+  std::string prefix;  ///< If there is no header or names, prepend this to the
+                       ///< column ID as the name
+  bool mangle_dupe_cols =
+      true;  ///< If true, duplicate columns get a suffix; if false, data will
+             ///< be overwritten if there are columns with duplicate names
+
+  bool dayfirst = false;  ///< Is the first value in the date formatthe day?
+                          ///< DD/MM  versus MM/DD
+
+  char thousands =
+      '\0';  ///< Numeric data thousands seperator; cannot match delimiter
+  char decimal = '.';   ///< Decimal point character; cannot match delimiter
+  char comment = '\0';  ///< Comment line start character; rest of the line will
+                        ///< not be parsed
+
+  char quotechar =
+      '\"';  ///< Character used to denote start and end of a quoted item
+  quote_style quoting =
+      QUOTE_MINIMAL;  ///< Treat string fields as quoted item and remove the
+                      ///< first and last quotechar
+  bool doublequote = true;  ///< Whether to interpret two consecutive quotechar
+                            ///< inside a field as a single quotechar
+
+  size_t byte_range_offset = 0;  ///< Bytes to skip from the start
+  size_t byte_range_size = 0;    ///< Bytes to read; always reads complete rows
+  gdf_time_unit out_time_unit =
+      TIME_UNIT_NONE;  ///< The output resolution for date32, date64, and
+                       ///< timestamp columns
+
+  RdKafka::Conf* kafka_configs;  ///< Defines global Kafka configurations that
+                                 ///< apply across an entire connected session.
+  std::vector<std::string>
+      kafka_topics;  ///< Topics that should be consumed from
+
+  int64_t start_offset;
```

Comment:
This structure already has an input byte offset - byte_range_offset. Please consider reusing it here instead of adding a new member

Reply:
I could use them but the fact that with Kafka I'm talking about a number of complete messages of arbitrary byte length gives me a little pause since the naming of those variables is clearly for a byte offset. That of course could be done but just might be a bit misleading to someone examining the code. Thoughts?

Reply:
I that case better to keep them separate.